package-system/AWSGameLiftServerSDK/docker_build_linux.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+
+
+WORKSPACE_DIR=/data/workspace
+SDK_SRC_SUBPATH=${1:-.}
+SRC_PATH=${WORKSPACE_DIR}/src/${SDK_SRC_SUBPATH}
+
+if [ ! -d ${SRC_PATH} ]
+then
+    echo "Unable to locate source path at ${SRC_PATH}"
+    exit 1
+fi
+
+# Perform a cmake project generation and build
+# 
+# Arguments:
+#   $1 : BUILD path
+#   $2 : Build SHARED libs
+#   $3 : Build Type
+build_package() {
+
+    echo "Generating $1"
+
+    cmake -G Ninja -S ${SRC_PATH} -B $1 -DBUILD_SHARED_LIBS=$2 -DCMAKE_BUILD_TYPE=$3
+    if [ $? -ne 0 ]
+    then
+        echo "Error generating AWS Gamelift Server SDK for $1" 
+        exit 1
+    fi
+    cmake --build $1
+    if [ $? -ne 0 ]
+    then
+        echo "Error building AWS Gamelift Server SDK for $1" 
+        exit 1
+    fi
+
+    if [ ! -d $1/prefix ]
+    then
+        echo "Error installing AWS Gamelift Server SDK for $1" 
+        exit 1
+    fi
+}
+
+BUILD_PATH_ROOT=${WORKSPACE_DIR}/build
+
+#### Build Static/Debug ####"
+echo "Building Static/Debug..."
+build_package ${BUILD_PATH_ROOT}/build_static_debug OFF Debug
+
+#### Build Static/Release ####
+echo "Building Static/Release ..."
+build_package ${BUILD_PATH_ROOT}/build_static_release OFF Release
+
+#### Build Shared/Debug ####
+echo "Building Shared/Debug..."
+build_package ${BUILD_PATH_ROOT}/build_shared_debug ON Debug
+
+#### Build Shared/Release ####
+echo "Building Shared/Release..."
+build_package ${BUILD_PATH_ROOT}/build_shared_release ON Release
```

Comment:
Why not use `"Ninja Multi-Config"` for the genrator and askew the need for `CMAKE_BUILD_TYPE`

Reply:
The default cmake for 20.04 is version 3.16, which doesn't support `Ninja Multi-Config` yet. I didn't think it was necessary to pull in an updated cmake into this image, considering i don't know how the gamelift source handles multi-config layouts

Reply:
Ok, that works.
However, why are we building debug configuration on Linux?

Reply:
I was debating whether or not to include the debug builds like the prior version did, but now I don't think its necessary. I will remove it and start this pattern for future linux 3P pacakges.